### ISSUE TITLE
False-positive: expected (), found String on inner else branch of nested if in let binding

### ISSUE THREAD

**Issue by UNIkeEN** [rust-lang/rust-analyzer#22915]
## Environment

- rust-analyzer: `0.4.2986-standalone (ca8fba2984 2026-07-24)` (pre-release)
- VS Code: `1.128.0`
- Extension: `rust-lang.rust-analyzer@0.4.2986`

## Summary

rust-analyzer emits a false-positive `expected (), found String` diagnostic on an inner `else` branch of a nested `if` inside a `let` binding expression, when the enclosing function returns `Result<(), _>`.

The inner `else` returns `String` — which is correct because it is a branch of the `let x = if … { … }` expression. rust-analyzer appears to confuse this branch with the function's return position.

## Minimal reproduction

```rust
fn toggle_mod(path: &str, enable: bool) -> Result<(), String> {
    let new_name = if enable {
        if let Some(_captures) = Some(path) {
            path.to_string()
        } else {
            path.to_string() // ❌ rust-analyzer: expected (), found String
        }
    } else {
        format!("{}.disabled", path)
    };
    println!("{}", new_name);
    Ok(())
}
```

`cargo check` compiles this without errors.

## Expected behavior

No diagnostic on the inner `else` branch — it correctly returns `String`, matching the type of the outer `if` expression assigned to `new_name`.

## Actual behavior

rust-analyzer reports: `expected (), found String` at the inner `else` branch, as if it were the function's return expression.

**Comment by ChayimFriedman2** [comment]
Given you didn't provide your Rust version (even though you need per the issue template), I cannot know for sure, but I'm pretty sure this is a duplicate.

Also note that [our AI policy](https://github.com/rust-lang/rust-analyzer/blob/master/AI_POLICY.md) disallows using AI to write GitHub comments, including issue descriptions.
